framework/src/main/java/org/checkerframework/framework/qual/PolymorphicQualifier.java

```diff
@@ -25,8 +25,10 @@
      * When multiple type hierarchies are supported by a single type system, then each polymorphic
      * qualifier needs to indicate which sub-hierarchy it belongs to. Do so by passing a qualifier
      * from the given hierarchy, by convention the top qualifier.
+     *
+     * @return the top qualifier in the hierarchy of this qualifier
      */
-    // We use the meaningless PolymorphicQualifier.class as default value and
+    // We use the meaningless Annotation.class as default value and
```

Comment:
Can you explain why using `Annotation.class` makes this clearer? Using `PolymorphicQualifier.class` makes it a bit clearer to me that this is about polymorphism... using just `Annotation.class` doesn't give any hint. Checking for either one shouldn't make a difference, does it? If it does, can you improve the comment?

Reply:
It doesn't matter what the default value is to the user.  The implementation is confusing because it is expecting the top qualifier, but PolymorphicQualifier is a meta-annotation.  It's also confusing to get the PolymorphicQualifier meta-annotation and then look for PolymorphicQualifier again.